### ISSUE TITLE
[Bug] Missing `user.crt` for user names longer than 65 characters

### ISSUE THREAD

**Issue by itadventurer** [strimzi/strimzi-kafka-operator#1694]
**Describe the bug**
When I create a user thats name is longer than 65 characters (this is the max) – the `user.crt` is missing in the secret

**To Reproduce**
1. Create a user with 65 characters (e.g. `my-test-onetwothreeforfivesixsevennineteneleventwelvethirteen.65`). Everything is fine – there is a `user.crt` in the secret
2. Create a user with 66 characters (e.g. `my-test-onetwothreeforfivesixsevennineteneleventwelvethirteen..66`). The `user.crt` is empty in the secret. Apart from that everyting seems to be fine

**Expected behavior**
`user.crt` is not missing

Example yaml:

```yaml
apiVersion: kafka.strimzi.io/v1alpha1
kind: KafkaUser
metadata:
  # 65 chars
  name: my-test-onetwothreeforfivesixsevennineteneleventwelvethirteen.65
  # 66 chars
  #name: my-test-onetwothreeforfivesixsevennineteneleventwelvethirteen..66
  labels:
    strimzi.io/cluster: kafka
spec:
  authentication:
    type: tls
  authorization:
    type: simple
    acls:
      - resource:
          type: topic
          name: foo
          patternType: prefix
        operation: Read
```

**Environment (please complete the following information):**
 - Strimzi version: 0.11.3
 - Installation method: YAML
 - Kubernetes cluster: OpenShift 3.9
 - Infrastructure: OpenStack

**Comment by scholzj** [comment]
Unfortunately, this problem seems to be caused by OpenSSL which has a limit for the CN part of the subject which is 64 characters. And the TLS users have the name of the KafkaUser resource as the CN in the subject of the TLS certificate. 

We should keep the issue opened and see whether we can maybe use something else to generate the certificates. But I’m afraid I have no quick solutions for this. :-(

**Comment by scholzj** [comment]
the error hndling around this should be fixed in #1698
